src/numbers/mod_int.rs

```diff
@@ -0,0 +1,100 @@
+//! # Modulo Integer
+
+use crate::numbers::extended_gcd::ext_gcd;
+
+pub const MOD: u64 = 998_244_353;
+
+pub struct Mint {
+    pub val: u64,
+}
+
+impl Mint {
+    pub fn new(val: u64) -> Self {
+        Self {
+            val: if val < MOD {
+                val
+            } else if val < 2 * MOD {
+                val - MOD
+            } else {
+                val % MOD
+            },
```

Comment:
I'm not a rustacean, but can you use a match here, or does that not work because it's an assignment? I tend to find the match a bit more readable than if/else, but this is preference. I do think there is value in using the features rust gives us though

Reply:
i'll switch to match